src/score.c

```diff
@@ -0,0 +1,171 @@
+        return 1;
+    }
+    else if(score1->points > score2->points){
+        return -1;
+    }
+    else{
+        return 0;
+    }
+}
+
+/*Read all scores from score file and returns a top_scores_t struct*/
+top_scores_t* read_scores(){
+    top_scores_t* topScores = (top_scores_t*) malloc(sizeof(top_scores_t));
+
+    FILE* fp = fopen(SCORES_FILE, "rb");
+    
+    if(fp == NULL){
+        topScores->nTopScores = 0;
+        return topScores;  
+    }
+
+    int nScores;
+    if(!fread(&nScores, sizeof(int), 1, fp)){
+        fclose(fp);
+        topScores->nTopScores = 0;
+        return topScores;  
+    }
+
+    topScores->nTopScores = nScores;
+
+    int i;
+    for(i = 0; i < nScores; i++) {
+        fread(topScores->scores[i].nickname, sizeof(char), MAX_NICKNAME + 1, fp);
+        fread(&(topScores->scores[i].points), sizeof(int), 1, fp);
+    }
+
+    fclose(fp);
+
+    return topScores;
+}
+
+
+/*Write top scores to score file*/
+void write_scores(top_scores_t* topScores){
+
+    FILE* fp = fopen(SCORES_FILE, "wb");
+    
+    if(fp == NULL){
+        return;
+    }
+
+    fwrite(&(topScores->nTopScores), sizeof(int), 1, fp);
+
+    int i;
+    for(i = 0; i < topScores->nTopScores; i++) {
+        fwrite(topScores->scores[i].nickname, sizeof(char), MAX_NICKNAME + 1, fp);
+        fwrite(&(topScores->scores[i].points), sizeof(int), 1, fp);
+    }
+
+    fclose(fp);
+
+    return;
+}
+
+/*Add new score to score file if has space or score is greater than last score*/
+void add_score(const char nickname[MAX_NICKNAME+1], int points){
+    score_t score;
+
+    strcpy(score.nickname, nickname);
+    score.points = points;
+
+    top_scores_t* topScores = read_scores();
+
+        
+    if(topScores->nTopScores == MAX_SCORES && score.points <= topScores->scores[MAX_SCORES-1].points){
+        free(topScores);
+        return;
+    }
+
+    if(topScores->nTopScores == MAX_SCORES) {
+        topScores->scores[topScores->nTopScores-1].points = score.points;
+
+        strcpy(topScores->scores[topScores->nTopScores-1].nickname, score.nickname);
+    }
+    
+    else {
+        topScores->scores[topScores->nTopScores].points = score.points;
+
+        strcpy(topScores->scores[topScores->nTopScores].nickname, score.nickname);
+        
+        topScores->nTopScores++;
+    }
+
+
+    qsort(topScores->scores,topScores->nTopScores,sizeof(score_t),compare_scores);
+
+    write_scores(topScores);
+
+    free(topScores);
+    return;
+}
+
+
+/*Print all score at the bottom of the window*/
+void print_scores(WINDOW* mainWindow, int NROWS, int NCOLS) {
+    top_scores_t* topScores = read_scores();
+
+    int i;
+
+    mvwprintw(mainWindow, NROWS*3/4+8,0,"Top Scores:\n");
+    int actualY = NROWS*3/4+10;
+    int actualX = 0;
+    for(i = 0; i < topScores->nTopScores; i++) {
+        mvwprintw(mainWindow,actualY,actualX,"%d: Nickname: %s; Score: %d\t", i+1, topScores->scores[i].nickname,  topScores->scores[i].points);
```

Comment:
Separating high scores with a tab ("\t") looks very weird to me (aesthetically). I would suggest to separate them with a "\n":
```suggestion
        mvwprintw(mainWindow,actualY,actualX,"%d: Nickname: %s; Score: %d\n", i+1, topScores->scores[i].nickname,  topScores->scores[i].points);
```
And, of course, replace lines 160-164 by a single incrementation of "actualY".

Reply:
```suggestion
        mvwprintw(mainWindow,actualY,actualX,"%d: Nickname: %4s; Score: %d\t", i+1, topScores->scores[i].nickname,  topScores->scores[i].points);

```

do you think it is better?

Reply:
This is definitely better, at least now there won't be any misaligned scores.